source/agora/utils/PrettyPrinter.d

```diff
@@ -411,6 +413,70 @@ GCOQ...LRIJ(61,000,000), GCOQ...LRIJ(61,000,000)`;
     assert(ResultStr == actual, actual);
 }
 
+/// Format array of blocks
+private struct BlocksFmt
+{
+    private const(Block[]) value;
+
+    public this (ref const Block[] r) @safe nothrow
+    {
+        this.value = r;
+    }
+
+    public void toString (scope void delegate (in char[]) @safe sink)
+        @safe nothrow
+    {
+        try
+        {
+            string sep1 = "\n===============================================================================\n";
+            formattedWrite(sink, "Blocks:%s", sep1);
+            formattedWrite(sink,
+                "%(%s\n====================\n%)",
+                this.value.map!(b => BlockFmt(b)));
```

Comment:
Since you are already using `map` for this, you could have made `BlocksFmt` templated and accept any range type. See one of my other comments how to do it.

Reply:
Now can be used for any inputRange.